source/agora/consensus/SlashPolicy.d

```diff
@@ -105,7 +105,8 @@ public class SlashPolicy
 
         foreach (idx, utxo_key; keys)
         {
-            if (!this.hasRevealedPreimage(height, utxo_key))
+            if (utxo_key != this.enroll_man.getEnrollmentKey() &&
+                !this.hasRevealedPreimage(height, utxo_key))
```

Comment:
How can `hasRevealedPreimage` return true for itself ? That shouldn't happen.

Reply:
I don't follow you. This call is getting whether the requested pre-images exists in the `ValidatorSet`.

Reply:
Yes, and it would make the code much simpler if `hasRevealedPreimage` would always returns `true` for `utxo_key == this.enroll_man.getEnrollmentKey()`. Because the node *knowns* the pre-images, so it should consider them "revealed".

Reply:
Got it, Thanks.